compiler/rustc_typeck/src/check/expr.rs

```diff
@@ -529,8 +533,13 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             _ => self.instantiate_value_path(segs, opt_ty, res, expr.span, expr.hir_id).0,
         };
 
-        if let ty::FnDef(..) = ty.kind() {
+        if let ty::FnDef(did, ..) = *ty.kind() {
             let fn_sig = ty.fn_sig(tcx);
+            if tcx.fn_sig(did).abi() == RustIntrinsic && tcx.item_name(did) == sym::transmute {
+                let from = fn_sig.inputs().skip_binder()[0];
+                let to = fn_sig.output().skip_binder();
+                self.deferred_transmute_checks.borrow_mut().push((from, to, expr.span));
```

Comment:
Again, why defer?

Reply:
Don't remember, probably inference vars, too. Will check and leave comments